### PR TITLE
Update Genotrance PX to latest (stable) release

### PR DESCRIPTION
**Pull request by FredericLespez**
This is a Pull request to update Genotrance PC the latest (stable) release.
https://github.com/genotrance/px/releases

### CODE REVIEW

File: bucket/px.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
     "homepage": "https://github.com/genotrance/px",
     "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy",
     "license": "MIT",
-    "version": "2018-05-18",
-    "url": "https://github.com/genotrance/px/releases/download/2018-05-18/px.exe",
-    "hash": "a57048602c634fdcf3ac9df8ba4c3e81856606ce5ea0dc156a33f6538e2c7574",
+    "version": "0.4.0",
+    "url": "https://github.com/genotrance/px/releases/download/v0.4.0/px.exe",
+    "hash": "93b54b5b6154c536f7ec0a85ed90ea25aa5ed66c7f61d2acbc37cab8a9231440",
     "bin": "px.exe"
```
Comment (Ash258) [inline bucket/px.json]:
```suggestion
    "bin": "px.exe",
    "checkver": "github",
    "autoupdate": {
        "url": "https://github.com/genotrance/px/releases/download/v$version/px.exe"
    }
```

File: bucket/px.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
     "homepage": "https://github.com/genotrance/px",
     "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy",
```
Comment (Ash258) [inline bucket/px.json]:
```suggestion
    "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy.",
```

**Comment by Ash258** [comment]
There is newer release tagged as vHed from May 20

**Comment by FredericLespez** [comment]
Thanks for your review.
If you prefer squashed commits, tell me so and i will rework the merge request.

**Comment by FredericLespez** [comment]
Sorry I didn't see your comment about the release.
Yes, there are newer releases. But they are pre-release. I aimed for the stable one.
The last one in particular is tagged vHEAD. I am afraid that the next pre-release will be tagged vHEAD and this one will be retagged with a date like all the previous pre-releases. If this happens, it will break installs done by scoop since the hash will change.

**Comment by Ash258** [comment]
> NOTE: Command line parameters passed with --install are not saved for use on startup. The --save flag or manual editing of px.ini is required to provide configuration to Px on startup.

https://github.com/genotrance/px#configuration

> Px requires only one piece of information in order to function - the server name and port of the proxy server. This needs to be configured in px.ini. 

Should be px.ini persisted??

**Comment by FredericLespez** [comment]
> Should be px.ini persisted??

I don't know since I don't use this feature. I will look into it tomorrow and I come back to you.

**Comment by FredericLespez** [comment]
> Should be px.ini persisted??

I looked into this issue today.
If you use the --install feature, you need to persist px.ini and px.ini should be located in the same directory as px.exe.

I tried to add a "persist": "px.ini" to the manifest, scoop tried to persist px.ini as a directory.
Is it possible to persist a file ?
Should we wait for the rework of data persistence (https://github.com/lukesampson/scoop/pull/3248) ?

File: bucket/px.json [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+    "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy.",
     "license": "MIT",
-    "version": "2018-05-18",
-    "url": "https://github.com/genotrance/px/releases/download/2018-05-18/px.exe",
-    "hash": "a57048602c634fdcf3ac9df8ba4c3e81856606ce5ea0dc156a33f6538e2c7574",
-    "bin": "px.exe"
+    "version": "0.4.0",
+    "url": "https://github.com/genotrance/px/releases/download/v0.4.0/px.exe",
+    "hash": "93b54b5b6154c536f7ec0a85ed90ea25aa5ed66c7f61d2acbc37cab8a9231440",
+    "bin": "px.exe",
```
Comment (Ash258) [inline bucket/px.json]:
```suggestion
    "bin": "px.exe",
    "persist": "px.ini",
```

File: bucket/px.json [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
-    "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy",
+    "description": "An HTTP proxy server to automatically authenticate through an NTLM proxy.",
     "license": "MIT",
-    "version": "2018-05-18",
-    "url": "https://github.com/genotrance/px/releases/download/2018-05-18/px.exe",
-    "hash": "a57048602c634fdcf3ac9df8ba4c3e81856606ce5ea0dc156a33f6538e2c7574",
-    "bin": "px.exe"
+    "version": "0.4.0",
+    "url": "https://github.com/genotrance/px/releases/download/v0.4.0/px.exe",
+    "hash": "93b54b5b6154c536f7ec0a85ed90ea25aa5ed66c7f61d2acbc37cab8a9231440",
```
Comment (Ash258) [inline bucket/px.json]:
```suggestion
    "hash": "93b54b5b6154c536f7ec0a85ed90ea25aa5ed66c7f61d2acbc37cab8a9231440",
    "pre_install": "if (-not (Test-Path \"$persist_dir\\px.ini\")) { New-Item \"$dir\\px.ini\" -ItemType File | Out-Null }",
```

**Comment by FredericLespez** [comment]
I have tested your suggestion : it works.
Thanks a lot for your help.

**Pull request merged**